dovecot: 2.3.0.1 -> 2.3.1

ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nix-update tools.

This update was made based on information from https://repology.org/metapackage/dovecot/versions.

These checks were done:

- built on NixOS
- ran `/nix/store/c20ip7wyymd39l7zisx38ky3bxp1sybv-dovecot-2.3.1/bin/dovecot --help` got 0 exit code
- ran `/nix/store/c20ip7wyymd39l7zisx38ky3bxp1sybv-dovecot-2.3.1/bin/dovecot --version` and found version 2.3.1
- found 2.3.1 with grep in /nix/store/c20ip7wyymd39l7zisx38ky3bxp1sybv-dovecot-2.3.1
- directory tree listing: https://gist.github.com/6d90467ee7649d7efc0a48eeacfc42c8

cc @viric @peti @rickynils @fpletz for review

ryantm committed:
dovecot: 2.3.0.1 -> 2.3.1

Semi-automatic update generated by https://github.com/ryantm/nix-update tools.

This update was made based on information from https://repology.org/metapackage/dovecot/versions.

These checks were done:

- built on NixOS
- ran `/nix/store/c20ip7wyymd39l7zisx38ky3bxp1sybv-dovecot-2.3.1/bin/dovecot --help` got 0 exit code
- ran `/nix/store/c20ip7wyymd39l7zisx38ky3bxp1sybv-dovecot-2.3.1/bin/dovecot --version` and found version 2.3.1
- found 2.3.1 with grep in /nix/store/c20ip7wyymd39l7zisx38ky3bxp1sybv-dovecot-2.3.1
- directory tree listing: https://gist.github.com/6d90467ee7649d7efc0a48eeacfc42c8

fpletz commented:
@GrahamcOfBorg build dovecot dovecotWithPgSQL dovecot_pigeonhole

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=532d3e30-ae2c-4962-802d-55f282cb248f)

Attempted: dovecot, dovecot_pigeonhole

The following builds were skipped because they don't evaluate on x86_64-linux: dovecotWithPgSQL

<details><summary>Partial log (click to expand)</summary><p>

```
libtool: link: gcc -shared  -fPIC -DPIC  .libs/managesieve-settings.o    -g -O2 -fstack-protector-strong   -Wl,-soname -Wl,libmanagesieve_settings.so -o .libs/libmanagesieve_settings.so
libtool: link: ( cd ".libs" && rm -f "libmanagesieve_settings.la" && ln -s "../libmanagesieve_settings.la" "libmanagesieve_settings.la" )
make[3]: Leaving directory '/tmp/nix-build-dovecot-pigeonhole-0.5.0.1.drv-0/dovecot-2.3-pigeonhole-0.5.0.1/src/managesieve'
make[2]: *** [Makefile:428: all-recursive] Error 1
make[2]: Leaving directory '/tmp/nix-build-dovecot-pigeonhole-0.5.0.1.drv-0/dovecot-2.3-pigeonhole-0.5.0.1/src'
make[1]: *** [Makefile:743: all-recursive] Error 1
make[1]: Leaving directory '/tmp/nix-build-dovecot-pigeonhole-0.5.0.1.drv-0/dovecot-2.3-pigeonhole-0.5.0.1'
make: *** [Makefile:602: all] Error 2
builder for ‘/nix/store/k37wmh0s8mk7rxzh4is8p8nmi4xgrwz6-dovecot-pigeonhole-0.5.0.1.drv’ failed with exit code 2
error: build of ‘/nix/store/k37wmh0s8mk7rxzh4is8p8nmi4xgrwz6-dovecot-pigeonhole-0.5.0.1.drv’ failed
```
</p></details>

fpletz committed:
dovecot_pigeonhole: 0.5.0.1 -> 0.5.1

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=38d1369b-9dd7-4d8f-936e-275a42f3bb6a)

The following builds were skipped because they don't evaluate on x86_64-darwin: dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=e023c957-69f9-4b8b-b350-2de3f4bfcd24)

Attempted: tests.dovecot

<details><summary>Partial log (click to expand)</summary><p>

```
building path(s) ‘/nix/store/0rad2ngfi2sfmmj4az3pbbs31wz7xcpd-etc’
building path(s) ‘/nix/store/rzfs9wyal48x7yy8g8wp5fz4vc8h0y8q-nixos-system-machine-18.09.git.98664f0’
building path(s) ‘/nix/store/alh5c6g727a4sjgh98xf48plawhgsdxc-closure-info’
/nix/store/aj9rz4k3xvg8q9fzqfnb0q123707zp0n-builder: line 1: .attrs.sh: No such file or directory
builder for ‘/nix/store/nxsawmj0narmm40j1g4n0mswk39g785k-closure-info.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/yr8wpmynmn9lf8s75ck8rwv32fbnfjv6-run-nixos-vm.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/r1h709xv1vq0s5wcim8d77wbjlmcm6g2-nixos-vm.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/nlfrmybdlhpdcfr2n9nn33hnhjml6mj0-nixos-test-driver-dovecot.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/65a0pqm73czada5lx1j0fy4bc6k2ynra-vm-test-run-dovecot.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/65a0pqm73czada5lx1j0fy4bc6k2ynra-vm-test-run-dovecot.drv’ failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=f59c8a67-97a4-4c5c-8abc-23a241fdf106)

Attempted: dovecot

The following builds were skipped because they don't evaluate on x86_64-darwin: dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=41c86fcc-120b-4284-a3e3-99a663d83e1b)

Attempted: dovecot, dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib/dovecot/settings/libmanagesieve_login_settings.so
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib/dovecot/settings/libmanagesieve_settings.so
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib/dovecot/settings/libpigeonhole_settings.so
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib/dovecot/lib90_sieve_plugin.so
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib/dovecot-2.3-pigeonhole/libdovecot-sieve.so.0.0.0
gzipping man pages under /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/share/man/
strip is /nix/store/fzcs0fn6bb04m82frhlb78nc03ny3w55-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib  /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/libexec  /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/bin
patching script interpreter paths in /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1
checking for references to /tmp/nix-build-dovecot-pigeonhole-0.5.1.drv-0 in /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1...
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=24c3b19c-a110-4cc3-a9ce-91cc7c5216ee)

Attempted: dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib/dovecot/settings/libmanagesieve_settings.so
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib/dovecot/settings/libmanagesieve_login_settings.so
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/libexec/dovecot/managesieve-login
shrinking /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/libexec/dovecot/managesieve
gzipping man pages under /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/share/man/
strip is /nix/store/fzcs0fn6bb04m82frhlb78nc03ny3w55-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/lib  /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/libexec  /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1/bin
patching script interpreter paths in /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1
checking for references to /build in /nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1...
/nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=3cdcac55-6f8c-4ed7-9c47-5693f932732d)

Attempted: dovecot, dovecot_pigeonhole

The following builds were skipped because they don't evaluate on aarch64-linux: dovecotWithPgSQL

<details><summary>Partial log (click to expand)</summary><p>

```
make[3]: *** [Makefile:838: managesieve-main.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/build/dovecot-2.3-pigeonhole-0.5.0.1/src/managesieve'
make[2]: *** [Makefile:428: all-recursive] Error 1
make[2]: Leaving directory '/build/dovecot-2.3-pigeonhole-0.5.0.1/src'
make[1]: *** [Makefile:743: all-recursive] Error 1
make[1]: Leaving directory '/build/dovecot-2.3-pigeonhole-0.5.0.1'
make: *** [Makefile:602: all] Error 2
builder for '/nix/store/052rcarv50rlg4a95si2ia4fqn4xmjx7-dovecot-pigeonhole-0.5.0.1.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/052rcarv50rlg4a95si2ia4fqn4xmjx7-dovecot-pigeonhole-0.5.0.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=865fbf08-7994-48a2-8dd3-66b2dc4fbaeb)

Attempted: dovecot, dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1/lib/dovecot/lib90_sieve_plugin.so
shrinking /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1/lib/dovecot/doveadm/lib10_doveadm_sieve_plugin.so
shrinking /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1/lib/dovecot-2.3-pigeonhole/libdovecot-sieve.so.0.0.0
gzipping man pages under /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1/share/man/
strip is /nix/store/3zq400fri5dv7d30lpxlqm2v9y1iis6j-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1/lib  /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1/libexec  /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1/bin
patching script interpreter paths in /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1
checking for references to /build in /nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1...
/nix/store/km3l6h477gwcifrjk32zd9lxjdx4afcz-dovecot-2.3.1
/nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=dda682ef-50cf-4ce2-9381-a81313ecd087)

Attempted: dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```
building '/nix/store/12jq68yfs16d8lvc26z26n0hk6lcf1h2-dovecot-2.3-pigeonhole-0.5.1.tar.gz.drv'...
waiting for locks or build slots...

trying http://pigeonhole.dovecot.org/releases/2.3/dovecot-2.3-pigeonhole-0.5.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   263  100   263    0     0    804      0 --:--:-- --:--:-- --:--:--   804
100 1750k  100 1750k    0     0  1305k      0  0:00:01  0:00:01 --:--:-- 3758k
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
/nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=386ea158-5030-4158-93e6-aa02aa7d766f)

Attempted: tests.dovecot

<details><summary>Partial log (click to expand)</summary><p>

```
machine: running command: systemctl --no-pager show "dovecot2.service"
machine: exit status 0
error: unit ‘dovecot2.service’ reached state ‘failed’
unit ‘dovecot2.service’ reached state ‘failed’
cleaning up
killing machine (pid 627)
vde_switch: EOF on stdin, cleaning up and exiting
vde_switch: Could not remove ctl dir '/build/vde1.ctl': Directory not empty
builder for '/nix/store/i906nj4drrrnsl41ww5797q9mpdqfmhf-vm-test-run-dovecot.drv' failed with exit code 255
[31;1merror:[0m build of '/nix/store/i906nj4drrrnsl41ww5797q9mpdqfmhf-vm-test-run-dovecot.drv' failed
```
</p></details>

fpletz committed:
nixos/dovecot: set group in config

The dovecot bump to 2.3.1 caused the dovecot service to fail to start
because it would try to chgrp sockets to dovecot whereas our default
dovecot group is called dovecot2.

fpletz commented:
The test error was a Nix2 compat issue but the dovecot test was indeed failing. Pushed a fix.

fpletz commented:
@GrahamcOfBorg test dovecot

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=234b89ac-e042-4594-b9b2-9ce3a8e3ed8a)

Attempted: tests.dovecot

<details><summary>Partial log (click to expand)</summary><p>

```
building path(s) ‘/nix/store/1wi432hvvf5ak691kgbafqiiijywx4v6-etc’
building path(s) ‘/nix/store/15ha5vwasfy1dlapnzrjb28p70imyaki-nixos-system-machine-18.09.git.35bb019’
building path(s) ‘/nix/store/8k6hfrli3dbbkwg41dlnr6qkzfj7n80q-closure-info’
/nix/store/aj9rz4k3xvg8q9fzqfnb0q123707zp0n-builder: line 1: .attrs.sh: No such file or directory
builder for ‘/nix/store/vqaq97f8hh42p5nkb438367jri55b7j0-closure-info.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/zjvvx9pm6ip6skgzv51jkbvizg1swaii-run-nixos-vm.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/ymf2iacm5xs4xw3ww8aiqbgcdr6c9585-nixos-vm.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/4d52dg2bfbncs2dgl76w1sdfax0lpf2j-nixos-test-driver-dovecot.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/p5pixcbgdiv701yrlyz1r9rihwxvhriz-vm-test-run-dovecot.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/p5pixcbgdiv701yrlyz1r9rihwxvhriz-vm-test-run-dovecot.drv’ failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=503563de-0b02-45a3-9f2b-4ad3eb918d1a)

Attempted: dovecot, dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/c20ip7wyymd39l7zisx38ky3bxp1sybv-dovecot-2.3.1
/nix/store/q6d29yr9fadp96k472mshazq566baaki-dovecot-pigeonhole-0.5.1
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=56d6eee4-b4c4-4ae0-9de7-14ec8e165ec1)

Attempted: dovecot, dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/km3l6h477gwcifrjk32zd9lxjdx4afcz-dovecot-2.3.1
/nix/store/c7kcgz6hyrmh6zdf5a4mh5l0zxffksmg-dovecot-pigeonhole-0.5.1
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38000&attempt_id=76963418-612f-4e8f-a5c5-d6a7ef5df446)

Attempted: dovecot

The following builds were skipped because they don't evaluate on x86_64-darwin: dovecot_pigeonhole

<details><summary>Partial log (click to expand)</summary><p>

```
make[2]: Leaving directory '/private/tmp/nix-build-dovecot-2.3.1.drv-0/dovecot-2.3.1/doc'
make[1]: Leaving directory '/private/tmp/nix-build-dovecot-2.3.1.drv-0/dovecot-2.3.1/doc'
post-installation fixup
gzipping man pages under /nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1/share/man/
strip is /nix/store/0fzpxnsanc02i4jsb1yhchjp4p62b2n3-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1/lib  /nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1/libexec  /nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1/bin  /nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1/sbin
patching script interpreter paths in /nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1
/nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1/libexec/dovecot/decode2text.sh: interpreter directive changed from "/bin/sh" to "/nix/store/x030a63qdilnv02pkivfjg44pdxsh5km-bash-4.4-p19/bin/sh"
moving /nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1/sbin/* to /nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1/bin
/nix/store/n3ygk44rlj063bcm3js1daqs6vyw4jih-dovecot-2.3.1
```
</p></details>
